File: src/main/java/ru/ssau/todo/entity/Task.java
package ru.ssau.todo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@RequiredArgsConstructor
@Table(name = "task")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "task_id")
    private long id;

    private String title;

    private String status;

    @ManyToOne
    @JoinColumn(name = "created_by", foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT))
    private User user;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
}

File: front-task/src/app/entity/Task.ts
import { TaskStatus } from "./TaskStatus";

export class Task{
    constructor(i: number, t: string, s: TaskStatus, cB: string, cA: Date){
        this.id = i;
        this.title = t;
        this.status = s;
        this.createdBy = cB;
        this.createdAt = cA;
    }
    
    id: number;
    title: string;
    status: TaskStatus;
    createdBy: string;
    createdAt: Date; 
  }